docs/content/en/docs/Advanced/build.md

```diff
@@ -279,3 +279,183 @@ gcloud compute images create <new_image_name> --source-uri=<your_bucket>/<cos-gc
+      nfs-utils \
+      open-iscsi \
+      rbd-nbd \
+      openrc \
+      openssh-client \
+      openssh-server \
+      parted \
+      procps \
+      qemu-guest-agent \
+      rng-tools \
+      rsync \
+      strace \
+      smartmontools \
+      sudo \
+      tar \
+      tzdata \
+      util-linux \
+      vim \
+      wireguard-tools \
+      wpa_supplicant \
+      xfsprogs \
+      xz \
+      open-vm-tools \
+      open-vm-tools-deploypkg \
+      open-vm-tools-guestinfo \
+      open-vm-tools-static \
+      open-vm-tools-vmbackup \
+      openntpd
+
+# Enable some services
+RUN rc-update add sshd boot && \
+    rc-update add connman boot  && \
+    rc-update add acpid boot && \
+    rc-update add hwclock boot && \
+    rc-update add syslog boot && \
+    rc-update add udev sysinit && \
+    rc-update add udev-trigger sysinit && \
+    rc-update add openntpd boot && \
+    rc-update add crond && \
+    rc-update add fail2ban
+
+# Symlinks to make boot-assessment work
+RUN ln -s /usr/sbin/grub-install /usr/sbin/grub2-install && \
+    ln -s /usr/bin/grub-editenv /usr/bin/grub2-editenv
+
+# Now we install Kairos dependencies below
+COPY --from=luet /usr/bin/luet /usr/bin/luet
+
+RUN mkdir -p /etc/luet/repos.conf.d && \
+    luet repo add kairos -y --url quay.io/kairos/packages --type docker
+
+RUN luet install -y \
+        system/base-cloud-config \
+        dracut/immutable-rootfs \
+        dracut/network \
+        static/grub-config \
+        system/suc-upgrade \
+        system/shim \
+        system/grub2-efi \
+        system/elemental-cli \
+        init-svc/openrc
+        # use init-svc/systemd for systemd based distros
+
+# Install kernels from the Kairos repositories, or regenerate the initrd with dracut
+RUN luet install -y distro-kernels/opensuse-leap distro-initrd/opensuse-leap
+
+# Enable services
+    RUN mkdir -p /etc/runlevels/default && \
+    ln -sf /etc/init.d/cos-setup-boot /etc/runlevels/default/cos-setup-boot  && \
+    ln -sf /etc/init.d/cos-setup-network /etc/runlevels/default/cos-setup-network  && \
+    ln -sf /etc/init.d/cos-setup-reconcile /etc/runlevels/default/cos-setup-reconcile && \
+    ln -sf /etc/init.d/kairos-agent /etc/runlevels/default/kairos-agent
+
+# systemd:
+	#RUN systemctl enable cos-setup-rootfs.service && \
+	#    systemctl enable cos-setup-initramfs.service && \
+	#    systemctl enable cos-setup-reconcile.timer && \
+	#    systemctl enable cos-setup-fs.service && \
+	#    systemctl enable cos-setup-boot.service && \
+	#    systemctl enable cos-setup-network.service
+```
+
+After building the container image, use the `osbuilder-tools` image to create an ISO:
+
+```bash
+docker run -v $PWD:/cOS -v /var/run/docker.sock:/var/run/docker.sock -i --rm quay.io/kairos/osbuilder-tools:latest --name "custom-iso" --debug build-iso --date=false --local $IMAGE --output /cOS/
+```
+
+Use QEMU to test the ISO:
+
+```bash
+qemu-system-x86_64 -m 2048 -drive if=virtio,media=disk,file=custom-iso.iso
```

Comment:
This may need some additional flags for network. I used a release opensuse iso (`kairos-alpine-opensuse-leap-v1.2.0.iso`) and the network interface didn't get an IP address so I couldn't connect to it (with the manual install, the QR code option went in a loop for some reason). I'll check what flags are there in qemu.

Reply:
My mistake, I didn't have the virtual network started.